packages/connection-form/src/components/advanced-options-tabs/csfle-tab/csfle-tab.tsx

```diff
@@ -0,0 +1,188 @@
+  fieldNameHasError,
+} from '../../../utils/validation';
+import type {
+  KMSProviderName,
+  KMSField,
+} from '../../../utils/csfle-kms-fields';
+import { KMSProviderFields } from '../../../utils/csfle-kms-fields';
+import FormFieldContainer from '../../form-field-container';
+
+const withMarginStyles = css({
+  marginTop: spacing[1],
+});
+
+const kmsProviderComponentWrapperStyles = css({
+  paddingLeft: spacing[3],
+  marginBottom: spacing[3],
+});
+
+interface KMSProviderMetadata {
+  kmsProvider: KMSProviderName;
+  title: string;
+  noTLS?: boolean;
+  clientCertIsOptional?: boolean;
+}
+
+const options: KMSProviderMetadata[] = [
+  {
+    title: 'Local KMS',
+    kmsProvider: 'local',
+    noTLS: true,
+  },
+  {
+    title: 'AWS',
+    kmsProvider: 'aws',
+  },
+  {
+    title: 'GCP',
+    kmsProvider: 'gcp',
+  },
+  {
+    title: 'Azure',
+    kmsProvider: 'azure',
+  },
+  {
+    title: 'KMIP',
+    kmsProvider: 'kmip',
+    clientCertIsOptional: false,
+  },
+];
+
+const containerStyles = css({
+  marginTop: spacing[3],
+});
+
+function CSFLETab({
+  connectionOptions,
+  updateConnectionFormField,
+  errors: errors_,
+}: {
+  errors: ConnectionFormError[];
+  updateConnectionFormField: UpdateConnectionFormField;
+  connectionOptions: ConnectionOptions;
+}): React.ReactElement {
+  const autoEncryptionOptions =
+    connectionOptions.fleOptions?.autoEncryption ?? {};
+
+  const errors = errorsByFieldTab(errors_, 'csfle');
+
+  const handleFieldChanged = useCallback(
+    (
+      key: keyof AutoEncryptionOptions,
+      value?: AutoEncryptionOptions[keyof AutoEncryptionOptions]
+    ) => {
+      return updateConnectionFormField({
+        type: 'update-csfle-param',
+        key: key,
+        value,
+      });
+    },
+    [updateConnectionFormField]
+  );
+
+  return (
+    <div className={containerStyles}>
+      <Banner>
+        Client-side Field-Level Encryption is an Enterprise/Atlas-only feature
+        of MongoDB.&nbsp;
+        <Link href="https://www.mongodb.com/docs/drivers/security/client-side-field-level-encryption-guide/">
+          Learn More
+        </Link>
+      </Banner>
+      <FormFieldContainer>
+        <TextInput
+          onChange={({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
+            handleFieldChanged('keyVaultNamespace', value);
+          }}
+          name="csfle-keyvault"
+          data-testid="csfle-keyvault"
+          label="Key Vault Namespace"
+          type="text"
+          optional={false}
+          value={autoEncryptionOptions.keyVaultNamespace || ''}
+          errorMessage={errorMessageByFieldName(errors, 'keyVaultNamespace')}
+          state={
+            fieldNameHasError(errors, 'keyVaultNamespace') ? 'error' : 'none'
+          }
+          spellCheck={false}
+          description="Specify a collection in which data encryption keys are stored in the format <db>.<collection>."
+        />
+      </FormFieldContainer>
+      <EncryptedFieldConfigInput
```

Comment:
How would you feel about wrapping this in the `FormFieldContainer` so that it gets a little bit of top margin?

We might be able to remove the first `withMarginStyles` from that component then 
https://github.com/mongodb-js/compass/pull/2928/files#diff-02ecb09ae0d4678618b443da67f6a9cdf5c3043d424727b39d494a2dc53fb9a9R55

Reply:
Sure, done – I’ve removed these everywhere and used `FormFieldContainer` instead. I’m not sure if this is what you had in mind exactly, but in any case these feel like things that are fairly easy to adjust :)